### PR TITLE
BXMSDOC-8071-main: Add Kafka and configmap sections

### PR DESCRIPTION
**Pull request by csherrar**

### CODE REVIEW

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-events-add-on.adoc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[id="con-kogito-events-add-on_{context}"]
+= {KOGITO} events add-on
+
+[role="_abstract"]
+The events add-on provides a default implementation in supported target platforms for `EventEmitter` and `EventReceiver` interfaces. You can use `EventEmitter` and `EventReceiver` interfaces to enable messaging by process, serverless workflow events, and event decision handling. For more information, see https://docs.jboss.org/kogito/release/latest/html_single/#con-knative-eventing_kogito-developing-process-services[_Knative Eventing in Kogito services_], https://docs.jboss.org/kogito/release/latest/html_single/#con-serverless-workflow-definitions_kogito-orchestrating-serverless[_Serverless Workflow definitions_], and {URL_GETTING_STARTED_KOGITO_MICROSERVICES}#proc-kogito-messaging-enabling_getting-started-kogito-microservices[_Enabling Kafka messaging for {KOGITO} services_].
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/con-kogito-events-add-on.adoc:5]: We do not have Serverless Workflow yet in product, thus I'd suggest removing the community links. 
```suggestion
The events add-on provides a default implementation in supported target platforms for `EventEmitter` and `EventReceiver` interfaces. You can use `EventEmitter` and `EventReceiver` interfaces to enable messaging by process, serverless workflow events, and event decision handling. 
```

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
+[id="con-kogito-operator-deployment-configs_{context}"]
+= {KOGITO} service properties configuration
+
+When a {KOGITO} service is deployed, a `configMap` resource is created for the `application.properties` configuration of the {KOGITO} service.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc:4]: In product, we call it microservice
```suggestion
When a {KOGITO} microservice is deployed, a `configMap` resource is created for the `application.properties` configuration of the {KOGITO} microservice.
```

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+The name of the `configMap` resource consists of the name of the {KOGITO} service and the suffix `-properties`, as shown in the following example:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc:6]:
```suggestion
The name of the `configMap` resource consists of the name of the {KOGITO} microservice and the suffix `-properties`, as shown in the following example:
```

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+.Example `configMap` resource generated during {KOGITO} service deployment
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc:8]:
```suggestion
.Example `configMap` resource generated during {KOGITO} microservice deployment
```

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+apiVersion: v1
+metadata:
+  name: kogito-travel-agency-properties
+data:
+  application.properties : |-
+    property1=value1
+    property2=value2
+----
+
+The `application.properties` data of the `configMap` resource is mounted in a volume to the container of the {KOGITO} service. Any runtime properties that you add to the `application.properties` section override the default application configuration properties of the {KOGITO} service.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc:21]:
```suggestion
The `application.properties` data of the `configMap` resource is mounted in a volume to the container of the {KOGITO} microservice. Any runtime properties that you add to the `application.properties` section override the default application configuration properties of the {KOGITO} microservice.
```

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+When the `application.properties` data of the `configMap` is changed, a rolling update modifies the deployment and configuration of the {KOGITO} service.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc:23]:
```suggestion
When the `application.properties` data of the `configMap` is changed, a rolling update modifies the deployment and configuration of the {KOGITO} microservice.
```

File: doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc [hunk 7]
```diff
@@ -0,0 +1,63 @@
+[id="proc-implementing-message-payload-decorator_{context}"]
+= Implementing message payload decorator
+
+[role="_abstract"]
+Any dependent add-on can implement the https://github.com/kiegroup/kogito-runtimes/blob/main/addons/common/messaging/common/src/main/java/org/kie/kogito/addon/cloudevents/message/MessagePayloadDecorator.java[_MessagePayloadDecorator_].
+
+.Prerequisites
+
+* You have installed the Events add-on in Kogito.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc:9]: Adding attribute for Red Hat build of Kogito.
```suggestion
* You have installed the Events add-on in {KOGITO}.
```

File: doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+. Create a file named `META-INF/services/org.kie.kogito.add-on.cloudevents.message.MessagePayloadDecorator` in your class path.
+
+. Open the file.
+
+. Enter the full name of your implementation class in the file.
+
+. Save the file.
++
+The `MessagePayloadDecoratorProvider` loads the file upon application start-up and adds the file to the decoration chain. When Kogito calls the `MessagePayloadDecoratorProvider#decorate`, your implementation is part of the decoration algorithm.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc:21]:
```suggestion
The `MessagePayloadDecoratorProvider` loads the file upon application start-up and adds the file to the decoration chain. When {KOGITO} calls the `MessagePayloadDecoratorProvider#decorate`, your implementation is part of the decoration algorithm.
```

File: doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+. To use the events add-on, add the following code to the pom.xml file of your project:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc:23]:
```suggestion
. To use the events add-on, add the following code to the `pom.xml` file of your project:
```

File: doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
++
+.Events smallrye add-on for Quarkus
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc:25]: We have an attribute for Red Hat build of Quarkus.
```suggestion
.Events smallrye add-on for {QUARKUS}
```

File: doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+[source,xml]
+----
+<dependency>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>kogito-addons-quarkus-events-smallrye</artifactId>
+  <version>1.15</version>
+</dependency>
+----
++
+.Events decisions add-on for Quarkus
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc:35]:
```suggestion
.Events decisions add-on for {QUARKUS}
```

File: doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc [hunk 12]
```diff
@@ -0,0 +1,63 @@
+[id="proc-implementing-message-payload-decorator_{context}"]
+= Implementing message payload decorator
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/getting-started-kogito/proc-implementing-message-payload-decorator.adoc:2]:
```suggestion
= Implementing message payload decorator for {KOGITO} events add-on
```

**Review by hmanwani-rh — COMMENTED** [review]
@csherrar Please see the comments below and let me know if you've any questions.

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-events-add-on.adoc (hunk 1 above)
Comment (csherrar) [reply doc-content/enterprise-only/getting-started-kogito/con-kogito-events-add-on.adoc:5]: Thanks! Done

File: doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc (hunk 2 above)
Comment (csherrar) [reply doc-content/enterprise-only/getting-started-kogito/con-kogito-operator-deployment-configs.adoc:4]: Done,and applying change to the rest of the PR

**Pull request merged**